### ISSUE TITLE
no-redundant-notify handle Subject.unsubscribe?

### ISSUE THREAD

**Issue by JasonWeinzierl** [JasonWeinzierl/eslint-plugin-rxjs-x#107]
If you call the `unsubscribe` method of a subject, does `no-redundant-notify` effect a failure?

**Comment by JasonWeinzierl** [comment]
Confirmed yes, the following code does not trigger an error but would always cause a runtime error:

```ts
        import { Subject } from "rxjs";
        const subject = new Subject<number>();
        subject.unsubscribe();
        subject.next(42);
```

`unsubscribe` is different than `complete` and `error`, and so the `no-redundant-notify` description doesn't exactly cover what `unsubscribe` does to a subject.  However since `no-subject-unsubscribe` is a recommended rule and thus users should typically not call `unsubscribe`, we're going to just add `unsubscribe` to `no-redundant-notify` instead of writing a new rule.
